contracts/TokenFactory.sol

```diff
@@ -326,18 +358,71 @@ contract TokenFactory is ReentrancyGuard, Ownable {
         }
     }
 
-    function burnTokenAndMintWinner(
+    function publishToUniswap (
         address tokenAddress
     ) external nonReentrant {
         uint256 _competitionId = competitionIds[tokenAddress];
+
+        require(_competitionId != currentCompetitionId, 'The competition is still active');
+
+        address winnerToken = getWinnerByCompetitionId(_competitionId);
+
+        require(winnerToken == tokenAddress, 'Token address not winner');
+
+        uint256 totalCollateral = getCollateralByCompetitionId(_competitionId);
+        WETH.deposit{value: totalCollateral}();
+
+        // calulate winner token amount
+        uint256 contributionWithoutFee = (totalCollateral * FEE_DENOMINATOR) /
+            (FEE_DENOMINATOR + feePercent);
+
+        uint256 winnerTokenAmount = bondingCurve.computeMintingAmountFromPrice(
+            totalCollateral,
+            Token(tokenAddress).totalSupply(),
+            contributionWithoutFee
+        );
+    
+        Token(tokenAddress).mint(address(this), winnerTokenAmount);
         
+        // token creator will get liquidity pull tokenId
+        (
+            address pool,
+            uint256 tokenId, 
+            uint128 liquidity, 
+            uint256 amount0, 
+            uint256 amount1
+        ) = tokenAddress < address(WETH) ? 
+            _addLiquidity(tokenAddress, winnerTokenAmount, address(WETH), totalCollateral, tokensCreators[tokenAddress]):
```

Comment:
This liquidity does not belong to `tokensCreators[tokenAddress]`. It belongs to the factory itself. Let's also make sure it has fee withdraw / WETH unwrap / token burning functions  for admin

Reply:
and token withdrawal - since the pool will earn fees in both WETH and the new token